components/asp/database/SystemWriter.py

```
@@ -57,9 +57,14 @@ def addDeployments(self, header):
         '''
         deployment_id = header.sysinfo["adliSystemExecutionId"]
         
-        system_id = header.sysinfo["metadata"]["systemId"]        
-        system_ver = header.sysinfo["metadata"]["systemVersion"]
-        system_ver = system_ver.replace(".","")
+        
+        if header.sysinfo and "metadata" in header.sysinfo and header.sysinfo["metadata"]:
+            system_id = header.sysinfo["metadata"]["systemId"] if "systemId" in header.sysinfo["metadata"] else None
```

Ruff (0.11.9)

Use `header.sysinfo["metadata"].get("systemId", None)` instead of an `if` block

Replace with `header.sysinfo["metadata"].get("systemId", None)`

(SIM401)